### PR TITLE
Create SimpleTimer class to wrap hardware and revise Timer

### PR DESCRIPTION
**Pull request by mikee47**
There are many instances where the extended capabilities of Timer are not required. Complex applications can use a fair number of these, each of which consumes more RAM than necessary for the task it is required to perform. The developer may therefore choose to 'downgrade' to an OSTimer to recover the RAM.

* Add OSTimer class as simple wrapper around SDK API
* Revise Timer class to use OSTimer

### CODE REVIEW

File: Sming/SmingCore/OSTimer.h [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ * short timer requirements are handled by an OSTimer. Still require Timer for
+ * longer periods and the additional callback flexibility.
+ *
+*/
+
+/** @defgroup   timer OSTimer functions
+ *  @brief      Provides timer functions
+*/
+
+#ifndef _SMING_CORE_OSTimer_H_
```
Comment (slaff) [inline Sming/SmingCore/OSTimer.h]: Use only capital letters for the guards.

File: Sming/SmingCore/OSTimer.h [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+/*
+ * According to documentation maximum value of interval for ms
+ * timer after doing system_timer_reinit is 268435ms.
+ * If we do some testing we find that higher values works,
+ * the actual limit seems to be about twice as high
+ * but we use the documented value anyway to be on the safe side.
+*/
+#define MAX_OS_TIMER_INTERVAL_US 268435000
+
+class OSTimer
```
Comment (slaff) [inline Sming/SmingCore/OSTimer.h]: Hm... Can we rename this to SimpleTimer ? I am not sure that even 2% of the developers will figure out which timer to prefer for their needs if they have the choice between OSTimer and Timer.

File: Sming/SmingCore/Timer.h [hunk 3]
```diff
@@ -12,16 +12,10 @@
 #define _SMING_CORE_Timer_H_
```
Comment (slaff) [inline Sming/SmingCore/Timer.h]: The guard should be in uppercase letters. I see now from where comes the "bad" influence.

File: Sming/SmingCore/Timer.h [hunk 4]
```diff
@@ -154,28 +164,28 @@ class Timer
...
      */
-	virtual void tick();
+	void tick();
 	/** @} */
 
 private:
-	os_timer_t timer;
-	uint64_t interval = 0;
+	OSTimer osTimer; ///< We use the OSTimer class to access the hardware
+	uint32_t interval = 0;
```
Comment (slaff) [inline Sming/SmingCore/Timer.h]: Is decreasing the timer precision from uint64_t to uint32_t correct?

Comment (mikee47) [reply Sming/SmingCore/Timer.h]: Yes, it only stores small (32-bit) counts, the longIntervalXXX variables deal with the rest. See `setIntervalUs()`.

File: Sming/SmingCore/Platform/OSTimer.h [hunk 5]
```diff
@@ -0,0 +1,101 @@
+/****
```
Comment (slaff) [inline Sming/SmingCore/Platform/OSTimer.h]: You have SimpleTimer declared two times. Once `Sming/SmingCore/Platform/OSTimer.h` and a second time in `Sming/SmingCore/SimpleTimer.h`. Maybe you should remove `Sming/SmingCore/Platform/OSTimer.h`.

File: Sming/SmingCore/SimpleTimer.h [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+ * short timer requirements are handled by an OSTimer. Still require Timer for
+ * longer periods and the additional callback flexibility.
+ *
+*/
+
+/** @defgroup   timer OSTimer functions
+ *  @brief      Provides timer functions
+*/
+
+#ifndef SMINGCORE_SIMPLETIMER_H_
```
Comment (slaff) [inline Sming/SmingCore/SimpleTimer.h]: Can you change `SMINGCORE_` to `_SMING_CORE_`?

File: Sming/SmingCore/Platform/OSTimer.h [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+ * Sming Framework Project - Open Source framework for high efficiency native ESP8266 development.
```
Comment (slaff) [inline Sming/SmingCore/Platform/OSTimer.h]: Offtopic: Please, don't mark review comments as resolved. Once you submit changes to PR the comments will collapse and I will know where to look more carefully and where to go briefly.

File: Sming/SmingCore/Platform/OSTimer.h (hunk 5 above)
Comment (mikee47) [reply Sming/SmingCore/Platform/OSTimer.h]: Finger trouble, fixed.

File: Sming/SmingCore/SimpleTimer.h (hunk 6 above)
Comment (mikee47) [reply Sming/SmingCore/SimpleTimer.h]: Have also fixed guard for Timer.h

**Pull request merged**